### PR TITLE
ENH: implement retrieval of atmospheric data from Windy API

### PR DESCRIPTION
**Pull request by Gui-FernandesBR**
## Pull request type

Please check the type of change your PR introduces:

- [x] Code base additions (bugfix, features)

## Pull request checklist

Please check if your PR fulfills the following requirements, depending on the type of PR:

- Code base additions (for bug fixes / features):

  - [ ] Tests for the changes have been added
  - [x] Docs have been reviewed and added / updated if needed
  - [x] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [x] All tests (`pytest --runslow`) have passed locally

## What is the new behavior?
@giovaniceotto is bringing windy API to improve our Environment models!

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
Can we skip the tests and examples additions? We can raise an issue saying that in the future we should describe, in environment calss usage noteook , how to use wind and, also, create unit tests

### CODE REVIEW

**Comment by giovaniceotto** [comment]
I do not agree with skipping changes to the Environment class notebook in this case. We already have a lot of issues opened. If we can, let's fix them before they are created.

**Comment by Gui-FernandesBR** [comment]
This is the only one lacking to go to develop before releasing new version. I can work on this by tomorrow. But please consider moving forward without this one if needed

**Comment by MateusStano** [comment]
Hey! I added the windy example to the usage notebook and changed a few things in the process. 

Just to be more in lines of what we were using in the code, I changed the inputs for the `file` argument to be able to be all capitalized. So before it had to be e.g. `ecmwf`, now it can also be `ECMWF`. In the documentation, I left the capitalized form as the standard.

Also, a big problem. The `iconEu` doesn't seem to be working, I couldn't really find the solution. Since we are not using all the atmospheric models available on windy, I think we could just remove `iconEu` if you are okay with it.

**Comment by giovaniceotto** [comment]
Quick note about iconEu, it only works if the latitude and longitude coordinates are within Europe, otherwise, the request returns `Out of model bounds`. Perhaps we can catch this error and notify the user?

**Comment by Gui-FernandesBR** [comment]
> Quick note about `ICON-EU`, it only works if the latitude and longitude coordinates are within Europe, otherwise, the request returns `Out of model bounds`. Perhaps we can catch this error and notify the user?

I think that's the best approach here

**Comment by MateusStano** [comment]
> Quick note about `ICON-EU`, it only works if the latitude and longitude coordinates are within Europe, otherwise, the request returns `Out of model bounds`. Perhaps we can catch this error and notify the user?

Done

**Review by Gui-FernandesBR — COMMENTED** [review]
This is my review, I think we are ready to merge this one. I cannot approve the PR since I was the author. @MateusStano or @giovaniceotto please one of you approve and they merge, 

tks for helping

File: rocketpy/Environment.py
Comment (giovaniceotto) [inline rocketpy/Environment.py:1463]:
```suggestion
                raise ValueError(
                    "Could not get a valid response for Icon-EU from Windy. Check if the latitude and longitude coordinates set are inside Europe.",
```

**Review by giovaniceotto — APPROVED** [review]

**Comment by giovaniceotto** [comment]
I approved this PR, but I found a single mistake which can easily be solved before merging:

image

It says that it will print the data for ECMWF but actually prints for ICON-EU

**Comment by MateusStano** [comment]
> I approved this PR, but I found a single mistake which can easily be solved before merging:
> 
> image
> 
> It says that it will print the data for ECMWF but actually prints for ICON-EU

Oh, good catch. Already fixed, going to merge now

**Pull request merged**
